src/components/SingleProduct.js

```diff
@@ -0,0 +1,30 @@
+import React, { useState, useEffect } from 'react';
+import { useParams } from 'react-router-dom';
+import Card from "react-bootstrap/Card";
+import Button from "react-bootstrap/Button";
+import { getProduct } from "../api";
+
+const SingleProduct = () => {
+    const [product, setProduct] = useState({});
+    const { productId } = useParams();
+
+    useEffect(() => {
```

Comment:
Do this!  Great! If you're at the single product route, fetch the product.  If this is mapping over the products, dont.